File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int l = 0;
        int r = numbers.size() - 1;
        while ( l < r) {
            int currentSum = numbers[l] + numbers[r];
            if ( currentSum > target ) r--;
            else if ( currentSum < target ) l++;
            else return {l+1, r+1};
        }
        return {};
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-2.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # we create a hashmap to map the value on the array to the index
        index = {}

        # now we populate the map
        for i, n in enumerate(nums): # enumerate gives (index, value)
            index[n] = i  

        for i, n in enumerate(nums):
            diff = target - n  # so we find the second number to deduce if that number is present in the array 
            if diff in index and index[diff] != i:
                return [i, index[diff]]